docs/source/tutorials/tutorial_tensordot_with_tc.rst

```diff
@@ -0,0 +1,117 @@
+----------------------------------------
+
+First step is to express the Tensordot operation in TC language. For more information on how to do
+so, you can refer to our `Documentation <https://facebookresearch.github.io/TensorComprehensions/index.html>`_
+and also find various TC examples `here <https://facebookresearch.github.io/TensorComprehensions/framework/pytorch_integration/layers_database.html>`_.
+
+.. code-block:: python
+
+    # import tc and torch both
+    import tensor_comprehensions as tc
+    import torch
+    # define the operation as TC language
+    lang = """
+    def tensordot(float(N, C1, C2, H, W) I0, float(N, C2, C3, H, W) I1) -> (O) {
+        O(n, c1, c3, h, w) +=! I0(n, c1, c2, h, w) * I1(n, c2, c3, h, w)
+    }
+    """
+
+Step 2: Register operation with TC
+----------------------------------
+
+Now, we will use the TC string and register it with the TC backend by calling :code:`tc.define`.
+
+.. code-block:: python
+
+    # register the lang with TC backend
+    tensordot = tc.define(lang, name="tensordot")
+
+.. note::
+
+    The :code:`name` variable should match the name of the def in the :code:`lang`.
+
+Step 3: Create input tensors and run TC
+---------------------------------------
+
+Now that TC is registered, we will create the input tensors and run it.
+
+.. code-block:: python
+
+    # create input cuda tensors
+    N, C1, C2, C3, H, W = 32, 512, 8, 2, 28, 28
+    I0, I1 = torch.randn(N, C1, C2, H, W).cuda(), torch.randn(N, C2, C3, H, W).cuda()
+    # choose the options that resemble the operation and run
+    out = tensordot(I0, I1, options=tc.Options("conv"))
+
+.. note::
+
+    In order to run the layer, we strongly recommend to pass the mapping options for
+    better performance. The options can also be obtained by autotuning the kernel
+    using Autotuner (next step) or you can chose defaults provided. See :ref:`must_pass_options`
+    for more information about options.
+
+Step 4: Autotune and get better performant kernel
+-------------------------------------------------
+
+So, it was very quick and easy to define the TensorDot operation with TC and get it running.
+But what about performance?
+
+TC provides a genetic algorithm based autotuner to tune the kernel performance. Let's
+autotune the kernel and get a better performance kernel. We will also cache the better
+kernel options by setting :code:`cache={filepath}` so that we can use these options
+later.
+
+.. code-block:: python
+
+    # autotune the kernel
+    best_options = tensordot.autotune(I0, I1, cache="tensordot_32_512_8_2_28.tc")
+    # run the kernel with the autotuned options
+    out = tensordot(I0, I1, options=best_options)
+
+You can control the amount of autotuning by changing the autotuner parameters. See
+:ref:`autotune_parameters` for how to change the settings.
+
+For the setting ``settings={"generations": 25, "pop_size": 100, "number_elites": 10}``, we
+get a decent kernel performance as shown in the screenshot below:
```

Comment:
How about the trick you did to abort the tuning when the performance is good enough?

Reply:
yeah, I should mention that too, let me add that as well :)